test/e2e/internal/notation/host.go

```diff
@@ -0,0 +1,109 @@
+package notation
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/notaryproject/notation/test/e2e/internal/utils"
+)
+
+// Host creates a virtualized notation testing host by modify
+// the "XDG_CONFIG_HOME" environment variable of the Executor.
+//
+// options is the required testing environment options
+// fn is the callback function containing the testing logic.
+func Host(options []utils.HostOption, fn func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost)) {
+	// create a vhost
+	vhost, err := utils.NewVirtualHost(NotationBinPath, CreateNotationDirOption())
+	if err != nil {
+		panic(err)
+	}
+
+	// set additional options
+	vhost.SetOption(options...)
+
+	// generate a repository with an artifact
+	artifact := GenerateArtifact()
+
+	// run the main logic
+	fn(vhost.Executor, artifact, vhost)
+
+	// remove the generated repository and artifact
+	if err := artifact.Remove(); err != nil {
+		panic(err)
+	}
+}
+
+// Opts is a grammar sugar to generate a list of HostOption
+func Opts(options ...utils.HostOption) []utils.HostOption {
+	return options
+}
+
+// BaseOptions returns a list of base Options for a valid notation
+// testing environment.
+func BaseOptions() []utils.HostOption {
+	return Opts(
+		AuthOption("", ""),
+		AddTestKeyOption(),
+		AddTestTrustStoreOption(),
+		AddTrustPolicyOption("trustpolicy.json"),
+	)
+}
+
+// CreateNotationDirOption creates the notation directory in temp user dir.
+func CreateNotationDirOption() utils.HostOption {
+	return func(vhost *utils.VirtualHost) error {
+		return os.MkdirAll(vhost.UserPath(NotationDirName), os.ModePerm)
+	}
+}
+
+// AuthOption sets the auth environment variables for notation.
+func AuthOption(username, password string) utils.HostOption {
+	if username == "" {
+		username = TestRegistry.Username
+	}
+	if password == "" {
+		password = TestRegistry.Password
+	}
+	return func(vhost *utils.VirtualHost) error {
+		vhost.UpdateEnv(authEnv(username, password))
+		return nil
+	}
+}
+
+// AddTestKeyOption adds the test signingkeys.json, key and cert files to
+// the notation directory.
+func AddTestKeyOption() utils.HostOption {
```

Comment:
we can remove `Test` from the name as all of functions are inside e2e folder.

Reply:
Updated.